### PR TITLE
Chore/linters

### PR DESCRIPTION
**Pull request by pepina-dev**
# Context
This repository currently lacks a linter configuration. Additionally, we have two lock files: one generated with Yarn and another with npm. To prevent potential issues arising from discrepancies between the lock files, we should standardize on a single package manager.

Furthermore, on small screens, the third logo in the sponsors' banner would look better if it were centered

# Changes
Adds  eslint, enforces yarn as oackage manager and changes deploy action to work with yarn and a custom domain. 

**Commit by commit**
- Adds eslint ans upgrades yarn to 4.5.3
- Fixes eslint offenses
- Adds github action to run eslint
- Removes package-lock.json to enforce use of yarn with yarn.lock. 
- Changes deploy action to work with custom domain. 
- Replaces scroll buttons that  go to the same page with a tags
- Centers the last logo in the sponsor banner 
- Updates README

**More info:**
- I tested the deploy in my fork with  a [personal apex domain](https://josefinalliende.com/)  and seems to be working fine. Some considerations to make this work. It is necessary to configure the following A, AAAA and CNAME records. The IPs for A and AAAA records are the ones listed in [this github docs](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-an-apex-domain) (See step 5 and section related to www. )
Screenshot 2024-12-01 at 9 44 59 PM
- After configuring this DNS records, github pages will generate de TLS certifcate with Let'sEncrypt and after that https can be enforced (this is already working ok for mostro.foundation)

- It was already configured correctly in this repo for A recordsn for the mostro.foudation domain, but AAAA config is missing ⚠
Screenshot 2024-12-03 at 3 29 52 PM

- Related to the logo in sponsors banner:

|Before|Now|
|----|----|
|Screenshot 2024-12-06 at 5 28 45 PM|Screenshot 2024-12-06 at 5 28 31 PM|

## Release Notes

- **New Features**
	- Introduced a new GitHub Actions workflow for automated linting of the codebase.
	- Added a deployment workflow for GitHub Pages with improved configurations.

- **Improvements**
	- Enhanced navigation elements across various components for better semantic structure.
	- Simplified layout in the sponsors banner for consistent alignment.
	- Streamlined the README to focus exclusively on `yarn` for dependency management.

- **Bug Fixes**
	- Corrected image tag structures to ensure proper HTML formatting.

- **Documentation**
	- Updated README with new title and focused setup instructions.
	- Added section for linters with relevant commands.

These changes enhance the overall user experience and maintain code quality.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -1,25 +1,36 @@
...
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
+      - run: corepack enable
+      - uses: actions/setup-node@v4
         with:
-          fetch-depth: 0
-      - uses: actions/setup-node@v3
+          node-version: v22.11
```
[the remark this answers is not in the document]
Comment (pepina-dev) [reply .github/workflows/deploy.yml:15]: node LTS is 22.11

File: package.json [hunk 2]
```diff
@@ -8,16 +8,19 @@
...
   "dependencies": {
     "nuxt": "^3.14.159",
     "vue": "latest",
     "vue-router": "latest"
   },
-  "packageManager": "yarn@3.6.1+sha512.de524adec81a6c3d7a26d936d439d2832e351cdfc5728f9d91f3fc85dd20b04391c038e9b4ecab11cae2b0dd9f0d55fd355af766bc5c1a7f8d25d96bb2a0b2ca",
   "devDependencies": {
+    "@nuxt/eslint": "^0.7.1",
     "@nuxtjs/tailwindcss": "^6.12.2",
+    "eslint": "^9.15.0",
```
[the remark this answers is not in the document]
Comment (pepina-dev) [reply package.json:23]: eslint 8 is EOL https://eslint.org/version-support/

File: package.json [hunk 3]
```diff
@@ -8,16 +8,19 @@
     "generate": "nuxt generate",
     "preview": "nuxt preview",
     "postinstall": "nuxt prepare",
-    "deploy": "gh-pages -d dist"
+    "deploy": "gh-pages -d dist",
+    "lint": "yarn eslint .",
+    "lintfix": "eslint . --fix"
```
[the remark this answers is not in the document]
Comment (pepina-dev) [reply package.json]: ✅  fixed!

File: components/hero-banner-header.vue [hunk 4]
```diff
@@ -32,8 +32,9 @@ import { smoothScroll } from '../utils/scroll'
           >
             About mostro
           </button>
-          <a @click="smoothScroll('#contact')"
-            class="text-sm/6 font-semibold text-white"
+          <a
+class="text-sm/6 font-semibold text-white"
+            @click="smoothScroll('#contact')"
```
[the remark this answers is not in the document]
Comment (pepina-dev) [reply components/hero-banner-header.vue]: ✅  changed to button. Edit: changed back to a tag but with href

File: nuxt.config.ts [hunk 5]
```diff
@@ -3,6 +3,7 @@ export default defineNuxtConfig({
   compatibilityDate: '2024-04-03',
   devtools: { enabled: true },
   modules: [
-    '@nuxtjs/tailwindcss'
+    '@nuxtjs/tailwindcss',
+    '@nuxt/eslint',
```
[the remark this answers is not in the document]
Comment (pepina-dev) [reply nuxt.config.ts:7]: ✅ Fixed!

**Comment by pepina-dev** [comment]
@coderabbitai review

File: .github/workflows/deploy.yml [hunk 6]
```diff
@@ -1,25 +1,36 @@
...
-permissions:
-  contents: write
-
 on:
+  workflow_dispatch:
   push:
     branches:
       - main
-
+      - chore/linters
```
[the remark this answers is not in the document]
Comment (pepina-dev) [reply .github/workflows/deploy.yml]: ✅  removed!

**Review by grunch — APPROVED** [review]
tACK

excelent job!

**Pull request merged**